test/e2e/common/kubelet.go

```diff
@@ -40,11 +40,11 @@ var _ = framework.KubeDescribe("Kubelet", func() {
 	Context("when scheduling a busybox command in a pod", func() {
 		podName := "busybox-scheduling-" + string(uuid.NewUUID())
 		/*
-			Release : v1.9
+			Release : v1.12
 			Testname: Kubelet, log output, default
 			Description: By default the stdout and stderr from the process being executed in a pod MUST be sent to the pod's logs.
 		*/
-		It("it should print the output to logs [NodeConformance]", func() {
+		framework.ConformanceIt("it should print the output to logs [NodeConformance]", func() {
```

Comment:
As I mentioned in #66969, I'm concerned about testing logs in conformance tests. Log durability is implementation-dependent (some implementations may aggressively aggregate and rotate logs), and longevity is scenario-dependent. 

Given that this is in NodeConformance, I gather it's used to validate container runtime implementations?

As for the tests that don't explicitly test logs but use it to convey pod output, could they use termination message instead?

Side note: That #1503 was effectively decided to be out of scope is one reason that this is hard to do in a portable fashion. Real applications would likely use an object store. We could write a ConfigMap, but that would require a client not included in busybox, as well as a service account authorized to write ConfigMaps. CRDs have similar issues, and also still beta.

Reply:
I'll follow up on the logs issue via email.

Reply:
> As for the tests that don't explicitly test logs but use it to convey pod output, could they use termination message instead?

I agree to emphasise the use of termiation message when it involves reading data from container and verify with expected output (specially in conformance space).

**Are you asserting** to tweek following 2 tests to enforce using termination message instead of reading pod's log? (as these e2e are mainly focused to convey pod output)
- Kubelet when scheduling a busybox Pod with hostAliases it should write entries to /etc/hosts [NodeConformance]
- Kubelet when scheduling a read only busybox container it should not write to root filesystem [NodeConformance]

Following 3 tests have different expression -
> Kubelet when scheduling a busybox command in a pod it should print the output to logs [NodeConformance]

Main dynamics of this test is to verify that container's data is written to pod's log. (should we give priority to use termination message instead of pod log here? )

> Kubelet when scheduling a busybox command that always fails in a pod should have an error terminated reason [NodeConformance]

Validates result against pod's data/output. (might not need any modification for promotion)

>Kubelet when scheduling a busybox command that always fails in a pod should be possible to delete [NodeConformance]

Just verifies if terminated pod is able to be deleted or not. (might not need any modification for promotion)

Reply:
@mgdevstack Thanks. Yes, I was suggesting such changes, but let's not make them now.

Let's first try to get the termination message tests promoted to conformance independently:
https://github.com/kubernetes/kubernetes/blob/81c6b735fa142657e24a25519d5db8a168dce7c2/test/e2e_node/runtime_conformance_test.go#L150

Reply:
Sure, thanks for pointing this out.